torch/csrc/jit/script/compiler.cpp

```diff
@@ -2318,7 +2318,7 @@ struct to_ir {
       Value* input,
       at::ArrayRef<Value*> indices) {
     auto* index =
-        graph->insertNode(graph->createList(DynamicType::get(), indices))
+        graph->insertNode(graph->createList(OptionalType::ofTensor(), indices))
```

Comment:
Please comment about why it is valid for this tensor to be optional.

Reply:
will leave a comment in the code. for here is to support removing the `createUndefined` when `emitIntAndSliceIndexing`, we don't create undefined tensor, instead we will insert Nones to the list, therefore we need to create list of optional tensors

Reply:
It is always List[Optional[Tensor]], but this comment makes it seem like it can be a List[Tensor], can you clarify it?

Reply:
This code is always creating a `List[Optional[Tensor]]`, the creation of this list is right on this line and it always has type List[Optional[Tensor]]

```
graph->insertNode(graph->createList(OptionalType::ofTensor(), indices))
```